crates/aptos-logger/src/aptos_logger.rs

```diff
@@ -751,12 +765,49 @@ fn json_format(entry: &LogEntry) -> Result<String, fmt::Error> {
     }
 }
 
+/// Periodically rebuilds the filter and replaces the current logger filter.
+/// This is useful for dynamically changing log levels at runtime via existing
+/// environment variables such as `RUST_LOG_TELEMETRY`.
+pub struct LoggerFilterUpdater {
+    logger: Arc<AptosData>,
+    logger_builder: AptosDataBuilder,
+}
+
+impl LoggerFilterUpdater {
+    pub fn new(logger: Arc<AptosData>, logger_builder: AptosDataBuilder) -> Self {
+        Self {
+            logger,
+            logger_builder,
+        }
+    }
+
+    pub async fn run(self) {
+        let mut interval = time::interval(FILTER_REFRESH_INTERVAL);
+        loop {
+            interval.tick().await;
+
+            self.update_filter();
+
+            info!("Logger filters rebuilt and reset.");
```

Comment:
This would be too noisy I guess, can we log this only when the filter changes or not log at all?